pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1470,6 +1470,45 @@ self: super: {
   pandoc-types = self.pandoc-types_1_21;
   rfc5051 = self.rfc5051_0_2;
 
+  # 2020-11-15: Taffybar and gi-* dependencies need these overrides to
+  # be compatible with haskell-gi-base 0.24.
+  taffybar = markUnbroken (appendPatch super.taffybar (pkgs.fetchpatch {
+    # Compatibility fix for haskell-gi-base 0.24
+    url = "https://github.com/taffybar/taffybar/pull/494/commits/a7443324a549617f04d49c6dfeaf53f945dc2b98.patch";
+    sha256 = "0prskimfpapgncwc8si51lf0zxkkdghn33y3ysjky9a82dsbhcqi";
+  }));
+  gi-cairo-connector = markUnbroken super.gi-cairo-connector;
+  gi-cairo-render = markUnbroken (overrideCabal (super.gi-cairo-render) (drv: {
+    # Fix for BoxedObject Change in haskell-gi 0.24
+    # https://github.com/taffybar/gi-cairo-render/pull/1
+    src = pkgs.fetchFromGitHub {
+      owner = "thestr4ng3r";
+      repo = "gi-cairo-render";
+      rev = "8727c43cdf91aeedffc9cb4c5575f56660a86399";
+      sha256 = "16kqh2ck0dad1l4m6q9xs5jqj9q0vgpqrzb2dc90jk8xwslmmhxd";
+    };
```

Comment:
Could you pull this in as a patch instead of updating the source here?

We generally prefer just adding patches, although I guess it is less of a problem on a release branch.

Reply:
I tried changing it to add the patch fetched from the PR, but the patch doesn't apply cleanly. However when digging around I noticed that master branch of the [taffybar/gi-cairo-render](https://github.com/taffybar/gi-cairo-render) fork contains the patch, and it builds too. I hope this is OK @cdepillabout .

Reply:
Hmm, I don't think I quite understand.  If `thestr4ng3r/gi-cairo-render` is forked from `taffybar/gi-cairo-render`, then is there something preventing you from taking the patch from the diff between them?

Reply:
To be honest, I don't quite understand either. There are multiple PRs and forks of this package. But it turns out there is a new hackage version which will work, so I have used that.

Reply:
Thanks, this looks much simpler now!